pynicotine/transfers.py

```diff
@@ -449,28 +449,32 @@ def file_limit_reached(self, user):
 
     def slot_limit_reached(self):
 
-        maxupslots = self.config.sections["transfers"]["uploadslots"]
+        uploadslots = self.config.sections["transfers"]["uploadslots"]
 
-        if maxupslots <= 0:
-            maxupslots = 1
+        if uploadslots <= 0:
+            uploadslots = 1
 
         in_progress_count = 0
-        now = time.time()
+        current_time = time.time()
 
-        for i in self.uploads:
+        for i in reversed(self.uploads):
             if i.conn is not None and i.speed is not None:
                 # Currently transferring
                 in_progress_count += 1
 
-            elif (now - i.last_status_change) < 30:
+            elif (current_time - i.last_status_change) < 30:  # ToDo: performance bottleneck
```

Comment:
I believe this time check is redundant today, because transfers time out after 30 seconds, and I don't think we have issues with stuck transfer statuses anymore.